File: src/main/java/com/everyonewaiter/domain/image/ImageFormat.java
package com.everyonewaiter.domain.image;

import com.sksamuel.scrimage.nio.ImageWriter;
import com.sksamuel.scrimage.nio.JpegWriter;
import com.sksamuel.scrimage.nio.PngWriter;
import com.sksamuel.scrimage.webp.WebpWriter;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.MediaType;

@Getter
@RequiredArgsConstructor
public enum ImageFormat {

  PNG("png", MediaType.IMAGE_PNG_VALUE, PngWriter.NoCompression),
  JPG("jpg", MediaType.IMAGE_JPEG_VALUE, JpegWriter.Default),
  WEBP("webp", "image/webp", WebpWriter.DEFAULT),
  ;

  private final String extension;
  private final String contentType;
  private final ImageWriter writer;

}

File: src/main/kotlin/com/everyonewaiter/common/file/converter/ImageFormat.kt
package com.everyonewaiter.common.file.converter

import com.sksamuel.scrimage.nio.ImageWriter
import com.sksamuel.scrimage.nio.JpegWriter
import com.sksamuel.scrimage.nio.PngWriter
import com.sksamuel.scrimage.webp.WebpWriter
import org.springframework.http.MediaType

/**
 * 이미지 포맷을 나타내는 열거형 클래스입니다.
 *
 * @property [extension] 이미지 포맷의 확장자
 * @property [contentType] 이미지 포맷의 Content-Type
 * @property [writer] 이미지 포맷을 쓰기 위한 ImageWriter
 * @see [com.sksamuel.scrimage.nio.ImageWriter]
 */
enum class ImageFormat(
    val extension: String,
    val contentType: String,
    val writer: ImageWriter,
) {
    PNG("png", MediaType.IMAGE_PNG_VALUE, PngWriter.NoCompression),
    JPG("jpg", MediaType.IMAGE_JPEG_VALUE, JpegWriter.Default),
    WEBP("webp", "image/webp", WebpWriter.DEFAULT),
}
